python/src/lib/workbook_validator.py

```diff
@@ -283,6 +284,9 @@ def validate_workbook(workbook: Workbook) -> Tuple[Errors, Optional[str]]:
             projects, subrecipients, version_string
         )
 
+    
+    subrecipients = [subrecipient.model_dump() for subrecipient in subrecipients]
```

Comment:
I think this is the "supported" way in the latest version of Pydantic, [see docs](https://docs.pydantic.dev/latest/concepts/serialization/#modelmodel_dump)

I tested by hardcoding the same `results` object we use in `validate_workbook` in `workbook_validator` and running unit tests, then inspecting the output. `json.dumps` works and subrecipient output looks good 
![Screenshot 2024-07-08 at 8 26 06 PM](https://github.com/usdigitalresponse/cpf-reporter/assets/19584337/49a8759e-4b5c-49d2-ad30-dc0b07cfc17a)